sandworm/lib/web.ml

```diff
@@ -1,63 +1,68 @@
-open Tyxml.Html
+module T = Tyxml.Html
 
-let bucket_url path =
-  Format.sprintf "https://dune-binary-distribution.s3.fr-par.scw.cloud/%s" path
-;;
-
-let title = title (txt "Dune Dev Preview")
-let css = link ~rel:[ `Stylesheet ] ~href:"main.css" ()
+let bucket_url ~url path = Format.sprintf "%s/%s" url path
+let title = T.title (T.txt "Dune Dev Preview")
+let css = T.link ~rel:[ `Stylesheet ] ~href:"main.css" ()
 
 let plausible_js =
-  script
+  T.script
     ~a:
-      [ Tyxml.Html.Unsafe.string_attrib "data-domain" "dune.ci.dev"
-      ; a_defer ()
-      ; a_src (Xml.uri_of_string "https://plausible.ci.dev/js/script.file-downloads.js")
+      [ T.Unsafe.string_attrib "data-domain" "dune.ci.dev"
+      ; T.a_defer ()
+      ; T.a_src
+          (T.Xml.uri_of_string "https://plausible.ci.dev/js/script.file-downloads.js")
       ]
-    (txt "")
+    (T.txt "")
 ;;
 
 let headers = [ css ]
-let main_title = h2 [ txt "Dune distribution binary" ]
+let main_title = T.h2 [ T.txt "Dune binary distribution" ]
 
 let install =
-  [ p
-      [ txt
-          "After downloaded and extracted, you could use the command below to install it."
+  [ T.p
+      [ T.txt "First, download the dune binary associated with your system requirement." ]
+  ; T.p
+      [ T.txt
+          "Then, you can install Dune by running the following command from the location \
+           where you downloaded the executable file:"
+      ]
+  ; T.pre [ T.code [ T.txt "$ sudo mv dune /usr/local/bin/dune" ] ]
+  ; T.p
+      [ T.txt
+          "Note that you can ignore this command and move the dune executable where you \
+           want, as long as it is accessible from the PATH."
       ]
-  ; pre [ code [ txt "$ sudo mv dune /usr/local/bin/dune" ] ]
-  ; p [ txt "Feel free to move the executable wherever you want!" ]
   ]
 ;;
 
-let target_html path target =
+let target_html ~url path target =
   let path = Fpath.(path / target / "dune" |> to_string) in
-  let url = bucket_url path |> Xml.uri_of_string in
+  let url = bucket_url ~url path |> T.Xml.uri_of_string in
   let name = Format.sprintf "dune-%s" target in
-  li [ a ~a:[ a_href url ] [ txt name ] ]
+  T.li [ T.a ~a:[ T.a_href url ] [ T.txt name ] ]
 ;;
 
-let bundle_html bundle =
+let bundle_html ~url bundle =
   let open Metadata.Bundle in
   let date = get_data_string_from bundle in
   let h3_title = Format.sprintf "Preview %s" date in
   let path = Fpath.v date in
   let targets = List.map Metadata.Target.to_string bundle.targets in
-  let aux acc target = target_html path target :: acc in
+  let aux acc target = target_html ~url path target :: acc in
   let targets = List.fold_left aux [] targets |> List.rev in
-  div [ h3 [ txt h3_title ]; ul targets ]
+  T.div [ T.h3 [ T.txt h3_title ]; T.ul targets ]
 ;;
 
-let content t =
-  let bundles = List.map bundle_html t in
+let content ~url t =
+  let bundles = List.map (bundle_html ~url) t in
   let body = (main_title :: install) @ bundles in
```

Comment:
I feel like this page would benefit from some text explaining the context for why it's necessary.

Reply:
Do you mean the generation part or the code in charge of building the bundles?

Reply:
Just generally what the binary distribution is and why one might want it. Imagine lacking the context, stumbling over the page (e.g. finding an old tab) and trying to understand what you are looking at.

Reply:
To be sure to understand: I should add more context on the generated page about why we have it?

Reply:
Given the page is not for us but for users, I think this is reasonable?

Reply:
Yes, yes, definitely! :+1: I was just asking to make sure I fully understand

Reply:
I have updated the code as you suggested: I also take the initiative to add @christinerose to the repository so she can make recommendations about the wording :+1: